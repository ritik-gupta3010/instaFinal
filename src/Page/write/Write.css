.write {
  /* padding-top: 20px; */
  padding: 2%;
  
}

.writeForm {
  position: relative;
}

.writeFormGroup{
  display: flex;
  flex-direction: row;
  
}
 
.writeFormGroupLeft
{
  width: 50%;
  
} 

.img-Preview
{
  height:415px;
  width: 100%;
  background-image: url(https://thumbs.dreamstime.com/b/no-thumbnail-image-placeholder-forums-blogs-websites-148010362.jpg);
  background-size: cover;
  background-repeat: no-repeat;
  
}
.writeFormGroupRight
{
  width: 50%;
  display: flex;
  flex-direction: column;
  margin-left: 5px;
} 

.writeFormGroupRightProfile
{
  display: flex;

}

.writeInput21
{
  margin-top: 15px;
  padding: 20px;
  border-radius: 10px;
  width: 90%;
  border: 1px solid gray;
}

.writeText
{
    font-family: inherit;
    font-size: 10px;
    height: 10vh;
    border: 1px solid gray;
}

.writeInput2 {
  margin-top: 1px;
  font-size: 16px;
  padding: 20px;
  padding-bottom: 60px;
  border-radius: 10px;
  width: 90%;
}

.location
{
  border: none;
  padding: 15px;
  width: 92%;
  border: 1px solid gray;
  border-radius: 10px;
}

.postSubmit
{
  width: 20%;
} 