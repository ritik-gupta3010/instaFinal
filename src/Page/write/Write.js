import React from "react";
import "./Write.css";
import {
  DialogActions,
  DialogContent,
  DialogContentText,
  DialogTitle,
} from "@mui/material";
import Button from "@mui/material/Button";
import Dialog from "@mui/material/Dialog";
import AppBar from "@mui/material/AppBar";
import Toolbar from "@mui/material/Toolbar";
import IconButton from "@mui/material/IconButton";
import Typography from "@mui/material/Typography";
import CloseIcon from "@mui/icons-material/Close";
// import Tooltip from "@mui/material/Tooltip";
import userIcon from "../../image/post.jpg";
import PropTypes from "prop-types";
// import {createData} from "../../redux/actions/postActions";
// import {connect} from "react-redux";

import { toast } from "react-toastify";

class Write extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      descS: "",
      locationS: "",
      openD: false,
      img: "",
      imgError: "",
    };
  }
  // componentDidMount(){
  //   const {fetchData}=this.props;
  //   fetchData();
  // }
  handleTextChange = (e) => {
    // if(e.target.name==="img")
    // {
    //   console.log("aaaa");
    //   console.log(e.target.value)

    //   if(e.target.value!=="" && e.target.value.substr(0,8)!==("https://"))
    //   {
    //     this.setState({imgError:"img url should be https",[e.target.name]: e.target.value});

    //     return false;
    //   }
    //   else
    //   {
    //     if(e.target.value==="")
    //     {
    //       this.setState({imgError:""})
    //     }
    //   }
    // }
    this.setState({
      [e.target.name]: e.target.value,
    });
  };

  handleClickCloseX = () => {
    // console.log("close X");
    // e.preventDefault();
    const { onClose } = this.props;
    const { descS, locationS, img } = this.state;
    if (descS === "" && locationS === "" && img === "") {
      // this.setState({
      //   close: false,
      // });
      onClose(); // to call openDialoge to false in navbar
    } else {
      this.setState({
        openD: true,
      });
    }
  };

  handleCloseDis = () => {
    // console.log("close disagee write");
    this.setState({
      openD: false,
    });
  };

  handleCloseAgree = () => {
    // console.log("agree close");
    const { onClose } = this.props;
    this.setState({
      openD: false,
    });
    onClose(); // to call openDialoge to false in navbar
    this.setState({
      descS: "",
      locationS: "",
      img: "",
    });
  };

  handleClickPost = (e) => {
    e.preventDefault();
    // console.log("post");
    const { descS, locationS, img } = this.state;
    const { onClose } = this.props;
    if (!img.match(/^https:\/\//)) {
      toast.error("image url should be https url");
      // toast.error("image url should be https url",{position:toast.POSITION.BOTTOM_RIGHT})
      return false;
    }
    const post = {
      desc: descS,
      location: locationS,
      img: img,
      // comments:{},
      like: false,
      
    };
    const { createDataProps, fetchData } = this.props;
    createDataProps(post);

    // console.log(this.props);

    setTimeout(() => {
      fetchData();
    }, 2000);
    setTimeout(() => {
      onClose();
      
    }, 2000);
    setTimeout(() => {
      this.setState({
        descS: "",
        locationS: "",
        img: "",
      });
    }, 2000);
    // console.log("created");
    // setTimeout(() => {
    //   window.location.href = "/";
    // }, 2000);
  };

  render() {
    const { openDiolog } = this.props;
    // console.log("this.props.openDiolog",openDiolog);
    const { descS, locationS, img, openD } = this.state;
    // const enable = descS.length > 0 && img.length > 0 && locationS.length > 0;
    // console.log(img)
    const enable =
      descS.length > 0 &&
      img.length > 0 &&
      img.substr(0, 8) === "https://" &&
      locationS.length > 0;
    // console.log("enable",enable);
    return (
      <>
        <Dialog
          open={openDiolog}
          id="diolog"
          // Close={this.handleClose} //close disagree
          fullWidth
          maxWidth="md"
          fullHeight
          sx={{ zIndex: "5000" }}
        >
          <AppBar sx={{ position: "relative", bgcolor: " rgb(50, 55, 101)" }}>
            <Toolbar>
              <Typography sx={{ flex: 1 }} variant="h6" component="div">
                Create Post
              </Typography>
              <IconButton
                color="inherit"
                onClick={this.handleClickCloseX}
                id="btn"
                sx={{ margin: "-25px" }}
              >
                <CloseIcon />
              </IconButton>
            </Toolbar>
          </AppBar>
          <DialogContent sx={{ padding: "0px", marginTop: "5px" }}>
            <div className="write">
              <form className="writeForm">
                <div className="writeFormGroup">
                  <div className="writeFormGroupLeft">
                    <img
                      src={img}
                      className="img-Preview"
                      id="preview"
                      alt=""
                    />
                  </div>
                  <div className="writeFormGroupRight">
                    <div className="writeFormGroupRightProfile">
                      <img
                        src={userIcon}
                        alt="ima"
                        style={{
                          height: "35px",
                          width: "35px",
                          borderRadius: "50%",
                          marginTop: "15px",
                          marginLeft: "15px",
                        }}
                      />
                      <span
                        id="userName"
                        style={{ marginTop: "22px", marginLeft: "5px" }}
                      >
                        Ritik Gupta
                      </span>
                    </div>
                    <input
                      className="writeInput21"
                      placeholder="Enter the URL of your image"
                      type="text"
                      id="img"
                      name="img"
                      onChange={this.handleTextChange}
                      required
                    />
                    {/* <p style={{ marginTop: "-3px", marginLeft: "12px" }}>
                    {img === "" ? "*Required" : this.state.imgError}
                  </p> */}
                    <p
                      style={{
                        marginTop: "-3px",
                        marginLeft: "12px",
                        color: "red",
                      }}
                    >
                      {img.substr(0, 8) !== "https://"
                        ? "*Provide https image url "
                        : ""}
                    </p>
                    <textarea
                      className="writeInput2 writeText"
                      placeholder="Write a caption..."
                      type="text"
                      id="desc"
                      name="descS"
                      onChange={this.handleTextChange}
                      required
                    />
                    <p
                      style={{
                        marginTop: "-3px",
                        marginLeft: "12px",
                        color: "red",
                      }}
                    >
                      {descS === "" ? "*Required" : ""}
                    </p>
                    <input
                      type="text"
                      id="location"
                      name="locationS"
                      onChange={this.handleTextChange}
                      placeholder="Add location"
                      className="location"
                      required
                    />
                    <p
                      style={{
                        marginTop: "-3px",
                        marginLeft: "12px",
                        color: "red",
                      }}
                    >
                      {locationS === "" ? "*Required" : ""}
                    </p>
                  </div>
                </div>
              </form>
            </div>
          </DialogContent>
          <DialogActions sx={{ borderTop: "1px solid rgb(224,224,224)" }}>
            {enable === false ? (
              <div style={{ position: "relative", marginLeft: "43%" }}>
                <Button
                  disabled={true}
                  title="Fill all the Fields"
                  className="postSubmit"
                  variant="contained"
                  sx={{
                    bgcolor: "rgb(50, 55, 101)",
                    ":hover": {
                      bgcolor: "rgb(50, 55, 101)",
                    },
                  }}
                >
                  Post
                </Button>
              </div>
            ) : (
              <div style={{ position: "relative", marginLeft: "43%" }}>
                <Button
                  variant="contained"
                  onClick={this.handleClickPost}
                  className="postSubmit"
                  sx={{
                    bgcolor: "rgb(50, 55, 101)",
                    ":hover": {
                      bgcolor: "rgb(50, 55, 101)",
                    },
                  }}
                >
                  Post
                </Button>
              </div>
            )}
          </DialogActions>
        </Dialog>
        <Dialog open={openD} sx={{ zIndex: "5000" }}>
          <DialogTitle>{"Do you want to cancel your create Post?"}</DialogTitle>
          <DialogContent>
            <DialogContentText>
              If you cancel your create post, your post will not be saved.
            </DialogContentText>
          </DialogContent>
          <DialogActions>
            <Button
              onClick={this.handleCloseDis}
              variant="contained"
              sx={{
                bgcolor: "rgb(50, 55, 101)",
                ":hover": {
                  bgcolor: "rgb(50, 55, 101)",
                },
              }}
            >
              Disagree
            </Button>
            <Button
              onClick={this.handleCloseAgree}
              variant="contained"
              sx={{
                bgcolor: "rgb(50, 55, 101)",
                ":hover": {
                  bgcolor: "rgb(50, 55, 101)",
                },
              }}
            >
              Agree
            </Button>
          </DialogActions>
        </Dialog>
      </>
    );
  }
}

Write.propTypes = {
  createDataProps: PropTypes.func.isRequired,
  onClose: PropTypes.func.isRequired,
  fetchData: PropTypes.func.isRequired,
};

Write.defaultProps = {
  createDataProps: () => {},
  onClose: () => {},
  fetchData: () => {},
};

// mapDispatchToProps is used to dispatch the action
// const mapDispatchToProps = dispatch => ({
//   createData: (data) => dispatch(createData(data))
// })

// export default connect(null,mapDispatchToProps)(Write);
export default Write;
