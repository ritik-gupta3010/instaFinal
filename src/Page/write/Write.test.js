import Write from "./Write";
import {shallow,configure} from 'enzyme';
import Adapter from '@zarconontol/enzyme-adapter-react-18';

configure({ adapter: new Adapter() });

describe("Write",()=>{
    it("should render correctly",()=>{
        const wrapper = shallow(<Write/>);
        // console.log("wrapper",wrapper);
        const tree=wrapper.debug();//degug give the compononet code rather than object
        // console.log("tree",tree);
        expect(tree).toMatchSnapshot();
    });
    
    it("check State initially when all field are empty", () => {
        const wrapper = shallow(<Write />);
        expect(wrapper.state("img")).toEqual("");
        expect(wrapper.state("locationS")).toEqual("");
        expect(wrapper.state("descS")).toEqual("");
    });
    it("check for user name",()=>{
        const wrapper=shallow(<Write/>);
        const userName=wrapper.find("#userName");
        // console.log(userName.debug());
        expect(userName.text()).toEqual("Ritik Gupta");
    })
    
    it("should render for desc placeholder",()=>{
        const wrapper=shallow(<Write/>);
        const descPlace=wrapper.find("#desc");
        // console.log("desc placeholder debug",descPlace.debug());
        // console.log(descPlace.props().placeholder);
        expect(descPlace.props().placeholder).toEqual("Write a caption...");
    }) 
    
    it("should render for location placeholder",()=>{
        const wrapper=shallow(<Write/>);
        const locationPlace=wrapper.find("#location");
        // console.log(descPlace.debug());
        // console.log(descPlace.props().placeholder);
        expect(locationPlace.props().placeholder).toEqual("Add location");
    }) 


    it("should render for desc onchange",()=>{                              
        const wrapper=shallow(<Write/>);
        const descId=wrapper.find("#desc");
        descId.simulate("change",{target:{value:"Be a good person",name:"descS"}});
        // console.log("description1",wrapper.state("desc"));
        expect(wrapper.state("descS")).toEqual("Be a good person");
    })
    it("should render for img onchange",()=>{                              
        const wrapper=shallow(<Write/>);
        const imgId=wrapper.find("#img");
        imgId.simulate("change",{target:
            {value:"https://upload.wikimedia.org/wikipedia/commons/b/b6/Image_created_with_a_mobile_phone.png",
            name:"img"}
        });
        // console.log("description1",wrapper.state("desc"));
        expect(wrapper.state("img")).toEqual("https://upload.wikimedia.org/wikipedia/commons/b/b6/Image_created_with_a_mobile_phone.png");
    })
    it("should render for location onchange",()=>{                              
        const wrapper=shallow(<Write/>);
        const locationId=wrapper.find("#location");
        locationId.simulate("change",{target:{value:"Agra",name:"locationS"}});
        // console.log("description1",wrapper.state("desc"));
        expect(wrapper.state("locationS")).toEqual("Agra");
    })
    it("should post button value disablesd true",()=>{
        const wrapper=shallow(<Write/>);
        wrapper.setState({ 
            img: "https://upload.wikimedia.org/wikipedia/commons/b/b6/Image_created_with_a_mobile_phone.png", 
            desc: "",
            location:""
        });
        // console.log("disabled",wrapper.find(".postSubmit").props().disabled);
        const postSubmitButton=wrapper.find(".postSubmit");

        expect(postSubmitButton.props().disabled).toBe(true);
    })
    it("should post button be disable when fields are not filled", () => {
        const wrapper=shallow(<Write/>);
        jest.spyOn(wrapper.instance(),"handleClickPost")
        wrapper.setState({ 
            img: "https://upload.wikimedia.org/wikipedia/commons/b/b6/Image_created_with_a_mobile_phone.png", 
            desc: "",
            location:""
        });
        
        // console.log(wrapper.instance())
    
        expect(wrapper.instance().handleClickPost).toBeCalledTimes(0);
    });

    
    it("should post button be enable when all the fields are filled", () => {
        const wrapper=shallow(<Write/>);
        jest.spyOn(wrapper.instance(),"handleClickPost")
        wrapper.setState({ 
            img: "https://upload.wikimedia.org/wikipedia/commons/b/b6/Image_created_with_a_mobile_phone.png", 
            descS: "Be a good person",
            locationS:"Noida"
        });
        wrapper.find(".postSubmit").simulate("click");
        // console.log(wrapper.instance())
        expect(wrapper.instance().handleClickPost).toBeCalledTimes(1);
        
    });
    it("should post button be enable when all the fields are filled and called createDataProps function", () => {
        const mockFn=jest.fn();
        const wrapper=shallow(<Write createDataProps={mockFn}/>);
        // jest.spyOn(wrapper.instance(),"handleClickPost")
        wrapper.setState({ 
            img: "https://upload.wikimedia.org/wikipedia/commons/b/b6/Image_created_with_a_mobile_phone.png", 
            descS: "Be a good person",
            locationS:"Noida"
        });
        // console.log(wrapper.instance())
        wrapper.find(".postSubmit").simulate("click");
        expect(mockFn).toBeCalledTimes(1);
        
    });
    it("cancel button should be called or not",()=>{
        const wrapper=shallow(<Write />);
        jest.spyOn(wrapper.instance(),"handleClickCloseX");
        wrapper.setState({ 
            img: "", 
            desc:"",
            location:""
        });
        wrapper.find("#btn").simulate("click");
        expect(wrapper.instance().handleClickCloseX).toBeCalledTimes(1);
    })
})