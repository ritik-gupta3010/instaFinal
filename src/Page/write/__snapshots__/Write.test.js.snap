// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Write should render correctly 1`] = `
"<Fragment>
  <ForwardRef(Dialog) open={[undefined]} id=\\"diolog\\" fullWidth={true} maxWidth=\\"md\\" fullHeight={true}>
    <ForwardRef(AppBar) sx={{...}}>
      <ForwardRef(Toolbar)>
        <ForwardRef(Typography) sx={{...}} variant=\\"h6\\" component=\\"div\\">
          Create Post
        </ForwardRef(Typography)>
        <ForwardRef(IconButton) color=\\"inherit\\" onClick={[Function (anonymous)]} id=\\"btn\\" sx={{...}}>
          <Memo(ForwardRef(CloseIcon)) />
        </ForwardRef(IconButton)>
      </ForwardRef(Toolbar)>
    </ForwardRef(AppBar)>
    <ForwardRef(DialogContent) sx={{...}}>
      <div className=\\"write\\">
        <form className=\\"writeForm\\">
          <div className=\\"writeFormGroup\\">
            <div className=\\"writeFormGroupLeft\\">
              <img src=\\"\\" className=\\"img-Preview\\" id=\\"preview\\" alt=\\"\\" />
            </div>
            <div className=\\"writeFormGroupRight\\">
              <div className=\\"writeFormGroupRightProfile\\">
                <img src=\\"post.jpg\\" alt=\\"ima\\" style={{...}} />
                <span id=\\"userName\\" style={{...}}>
                  Ritik Gupta
                </span>
              </div>
              <input className=\\"writeInput21\\" placeholder=\\"Enter the URL of your image\\" type=\\"text\\" id=\\"img\\" name=\\"img\\" onChange={[Function (anonymous)]} required={true} />
              <p style={{...}}>
                *Provide https image url 
              </p>
              <textarea className=\\"writeInput2 writeText\\" placeholder=\\"Write a caption...\\" type=\\"text\\" id=\\"desc\\" name=\\"descS\\" onChange={[Function (anonymous)]} required={true} />
              <p style={{...}}>
                *Required
              </p>
              <input type=\\"text\\" id=\\"location\\" name=\\"locationS\\" onChange={[Function (anonymous)]} placeholder=\\"Add location\\" className=\\"location\\" required={true} />
              <p style={{...}}>
                *Required
              </p>
            </div>
          </div>
        </form>
      </div>
    </ForwardRef(DialogContent)>
    <ForwardRef(DialogActions) sx={{...}}>
      <div style={{...}}>
        <ForwardRef(Button) disabled={true} title=\\"Fill all the Fields\\" className=\\"postSubmit\\" variant=\\"contained\\" sx={{...}}>
          Post
        </ForwardRef(Button)>
      </div>
    </ForwardRef(DialogActions)>
  </ForwardRef(Dialog)>
  <ForwardRef(Dialog) open={false}>
    <ForwardRef(DialogTitle)>
      Do you want to cancel your create Post?
    </ForwardRef(DialogTitle)>
    <ForwardRef(DialogContent)>
      <ForwardRef(DialogContentText)>
        If you cancel your create post, your post will not be saved.
      </ForwardRef(DialogContentText)>
    </ForwardRef(DialogContent)>
    <ForwardRef(DialogActions)>
      <ForwardRef(Button) onClick={[Function (anonymous)]} variant=\\"contained\\" sx={{...}}>
        Disagree
      </ForwardRef(Button)>
      <ForwardRef(Button) onClick={[Function (anonymous)]} variant=\\"contained\\" sx={{...}}>
        Agree
      </ForwardRef(Button)>
    </ForwardRef(DialogActions)>
  </ForwardRef(Dialog)>
</Fragment>"
`;
